src/hotspot/share/runtime/arguments.cpp

```diff
@@ -2952,6 +2955,37 @@ jint Arguments::parse_each_vm_init_arg(const JavaVMInitArgs* args, bool* patch_m
     LogConfiguration::configure_stdout(LogLevel::Info, true, LOG_TAGS(class, path));
   }
 
+  // This must be done after all options have been processed
+  // since -Xlog and -Xloggc override corresponding -verbose
+  for (LogTagSet* ts = LogTagSet::first(); ts != NULL; ts = ts->next()) {
+    if (ts->contains(LogTag::_gc) && ts->has_output_to_file()) {
+      has_verbose_gc = false;
+    }
+    if (ts->contains(LogTag::_class) && ts->has_output_to_file()) {
+      has_verbose_class = false;
```

Comment:
This doesn't seem correct to me. If the command line is:

```
java -Xlog:class+resolve=debug:file=log.txt -verbose:class -version
```

We should have the following:

```
(a) class,load -> stdout
(b) class,unload -> stdout
(c) class,resolve -> log.txt
```

But your change will disable both (a) and (b).


Also, what is the proposed behavior with this?

```
java -Xlog:class+load:file=log.txt -verbose:class -version
```

Will we have [A]:

```
class,load -> log.txt
class,unload -> stdout
```

or [B]?

```
class,load -> log.txt
class,unload -> disabled
```

Reply:
For your java command:
```
java -Xlog:class+load:file=log.txt -verbose:class -version
```
We will have [B] according to the current commit:
```
class,load -> log.txt
class,unload -> disabled
```
But I'm not sure whether this behavior is correct. The website (https://docs.oracle.com/en/java/javase/16/docs/specs/man/java.html) doesn't give us explicit clue. It only says -verbose:class equals -Xlog:class+load=info,class+unload=info.
For the command above, maybe we can propose a third way [C]:
```
class,load -> log.txt and stdout
class,unload -> stdout
```
In addition, what if the command is:
```
java -Xlog:class+load=error,class+unload=info:log.txt -verbose:class -version
```
In this case, since ``class+load=error`` is ”weaker" than ``class+load=info``, maybe -verbose should not be overriden by -Xlog?
To make things simple, in my commit, as long as "class"  shows up, no matter it's ``-Xlog:class`` or ``-Xlog:class+resolve`` or ``-Xlog:class=debug``  , then ``-verbose:class`` is to be overiden. But I'm quite uncertain about whether it is correct. Looking forward to your suggestions.